### PR TITLE
ARROW-8444: [Documentation] Fix spelling errors across the codebase

### PR DESCRIPTION
**Pull request by kszucs**
Quickly run `codespell` which found a couple of misspellings.

### CODE REVIEW

File: cpp/src/arrow/type.h [hunk 1]
```diff
@@ -1652,7 +1652,7 @@ class ARROW_EXPORT Schema : public detail::Fingerprintable,
   /// \brief Return the number of fields (columns) in the schema
   int num_fields() const;
 
-  /// Return the ith schema element. Does not boundscheck
+  /// Return the with schema element. Does not boundscheck
```
Comment (bkietz) [inline cpp/src/arrow/type.h:1655]: This is incorrect- it was intended to be `ith` as in the element at position `i`

**Review by bkietz — CHANGES_REQUESTED** [review]
Thanks for doing this! Just one spurious change

File: cpp/src/arrow/type.h (hunk 1 above)
Comment (kszucs) [reply cpp/src/arrow/type.h]: @bkietz Indeed, updated.

Comment (pitrou) [reply cpp/src/arrow/type.h]: Perhaps "i-th" to make it less ambiguous?

**Review by bkietz — APPROVED** [review]

**Pull request closed without merging**